### PR TITLE
Add SAFETY comments to core::slice::sort::partition_in_blocks

### PR DESCRIPTION
**Pull request by mdsn**
A few more SAFETY comments for #66219. There are still a few more in this module.

@rustbot label T-libs T-compiler C-cleanup

### CODE REVIEW

File: library/core/src/slice/sort.rs [hunk 1]
```diff
@@ -389,11 +405,20 @@ where
...
+            // safe. Otherwise, the debug assertions above guarantee that `width(l, r) == block_l + block_r`,
+            // namely, that the block sizes have been adjusted to account for the smaller number of remaining
+            // elements.
             l = unsafe { l.offset(block_l as isize) };
         }
 
         if start_r == end_r {
             // All out-of-order elements in the right block were moved. Move to the previous block.
+
+            // SAFETY: Same argument as above. Either this is a full block `2*BLOCK`-wide, or `block_r` has
```
Comment (poliorcetics) [inline library/core/src/slice/sort.rs:420]: Just a small nit: if can be unclear if this refers to the previous comment or the previous Safety comment. I don't have a link but you can "name" the referred comment something like this:

```
// a-somewhat-logical-name
// SAFETY: expl

...

// SAFETY: expl
// also refer to comment [a-somewhat-logical-name] for more reasons
```

**Review by poliorcetics — COMMENTED** [review]
Welcome and thanks for the PR ! I hope we can finally close the overall issue with this one

**Comment by mdsn** [comment]
Sorry @Mark-Simulacrum I didn't see you approved the PR before and just force-pushed a squashed single commit. Will this cause a problem?

**Comment by Mark-Simulacrum** [comment]
@bors r+ rollup

Thanks! Just need to reapprove, no problem.

**Pull request merged**